opus/application/static_media/js/browse.js

```diff
@@ -1196,6 +1200,53 @@ var o_browse = {
         return url;
     },
 
+    // check the cache of both rendered elements and variable o_browse.galleryData; remove 'far away' observations
+    // from cache to avoid buildup of too much data in the browser which slows things down
+    // Two functions; one to delete single elements, just a tool for the main one, manageObservationCache, to loop.
+    manageObservationCache: function(count, append) {
+        let tab = `#${opus.prefs.view}`;
+        var galleryObsElem =  $(`${tab} .gallery [data-obs]`);
+        var tableObsElem = $(`${tab} .op-dataTable-view [data-obs]`);
+
+        function deleteCachedObservation(index) {
+            // don't delete the metadata if the observation is in the cart
+            if (!galleryObsElem.eq(index).hasClass("in")) {
+                let delOpusId = galleryObsElem.eq(index).data("id");
+                delete o_browse.galleryData[delOpusId];
+            }
+            galleryObsElem.eq(index).remove();
+            tableObsElem.eq(index).remove();
+        };
+
+        let beginningIndex = galleryObsElem.first().index();
+        let endingIndex = galleryObsElem.last().index();
+        if (beginningIndex < 0 || endingIndex < 0) {
+            // this only happens when there are no elements rendered, so why bother...
+            // probably don't need to check both, but why not...
+            return;
+        }
+
+        let totalObs = endingIndex - beginningIndex;
```

Comment:
Actually total # of observations is ending - beginning + 1, right?

Reply:
nope; i changed this so it's 0-based when i changed to 'index' vs. 'obsNumber'.

Reply:
It doesn't matter if it's zero-based. End-start is still off by 1 for the total number of obs.
0 1 2 3
3-0=3 but there are 4 obs